src/main/java/com/salesforce/dataloader/controller/Controller.java

```diff
@@ -462,24 +465,24 @@ public static synchronized void initLog() throws FactoryConfigurationError {
             return;
         }
 
-        try {
-            File logConfXml = Paths.get(System.getProperty("user.dir"), "configs", LOG_CONF_OVERRIDE).toFile();
-            if (logConfXml.exists()) {
-                logger.info("Reading log-conf.xml in " + logConfXml.getAbsolutePath());
-                if (logConfXml.canRead()) {
-                    DOMConfigurator.configure(logConfXml.getAbsolutePath());
-                } else {
-                    logger.warn("Unable to read log-conf.xml in " + logConfXml.getAbsolutePath());
-                }
-            } else {
-                logger.info("Using built-in logging configuration, no log-conf.xml in " + logConfXml.getAbsolutePath());
-            }
-            logger.info(Messages.getString("Controller.logInit")); //$NON-NLS-1$
-        } catch (Exception ex) {
-            logger.error(Messages.getString("Controller.errorLogInit")); //$NON-NLS-1$
-            logger.error(ex);
-            System.exit(1);
+        File logConfXml = Paths.get(System.getProperty("user.dir"), "configs", LOG_CONF_OVERRIDE).toFile();
+        if (logConfXml.exists()) {
+        	System.setProperty("log4j.configurationFile", logConfXml.getAbsolutePath());
+        } else {
+            System.out.println("Using built-in logging configuration, no log-conf.xml in " + logConfXml.getAbsolutePath());
```

Comment:
I removed the else clause because the logger is not initialize at this point. A check after initialization of the logger makes the following call if logger wasn't initialized from the config file:
if (logConfigLocation == null) {
        	logger.error("Unable to initialize logging using log4j2 config file at "
        			+ logConfXml.getAbsolutePath()
        			+ ". All error messages will be logged on STDOUT.");
        } else {
        	logger.info("Using log4j2 configuration file at location: " + logConfigLocation);
        }